src/main/java/com/uber/cadence/migration/MigrationInterceptor.java

```diff
@@ -70,8 +71,10 @@ public byte[] executeWorkflow(
         // Skip migration on non-cron and child workflows
         WorkflowExecutionStartedEventAttributes startedEventAttributes =
             input.getWorkflowExecutionStartedEventAttributes();
-        if (startedEventAttributes.cronSchedule == ""
-            || startedEventAttributes.getParentWorkflowExecution().getWorkflowId() != "") {
+        if (startedEventAttributes.cronSchedule == null
+            || !startedEventAttributes.cronSchedule.equals(""))
```

Comment:
incorrect logic and it might be easier to have `isCronWorkflow` method
```suggestion
            || startedEventAttributes.cronSchedule.equals(""))
```